doc/src/modules/matrices/random.rst

```diff
@@ -0,0 +1,222 @@
+    *isometry conjugate matrices* that arise as a product $\mathbf{OAO}^{-1} = \mathbf{OAO}^{T}$
+    of specific matrices $A$
+    where $\mathbf{A}$ sets the (complex) eigenvalue spectrum.
+
+    - :py:func:`orthogonal <sympy.matrices.random.orthogonal>`
+      can be a conjugate of
+      :py:func:`isometry_normal <sympy.matrices.random.isometry_normal>`
+      by
+      :py:func:`orthogonal <sympy.matrices.random.orthogonal>`
+
+    - same holds for :py:func:`unitary <sympy.matrices.random.unitary>`
+      which can be a conjugate of
+      :py:func:`isometry_normal <sympy.matrices.random.isometry_normal>`
+      by
+      :py:func:`unitary <sympy.matrices.random.unitary>`
+
+    - :py:func:`normal <sympy.matrices.random.normal>`
+      is a conjugate of
+      :py:func:`diagonal_normal <sympy.matrices.random.diagonal_normal>`
+      by either
+      :py:func:`orthogonal <sympy.matrices.random.orthogonal>`
+      or
+      :py:func:`unitary <sympy.matrices.random.unitary>`
+
+    finally *symmetric matrices* given as a as product $\mathbf{MM}^T$
+    resp. $\mathbf{MM}^H$ of $\mathbf{M}$, such as
+
+    - :py:func:`symmetric <sympy.matrices.random.symmetric>` $\mathbf{MM}^T$ with
+      :py:func:`invertible <sympy.matrices.random.invertible>` $\mathbf{M}$
+
+    - :py:func:`hermite <sympy.matrices.random.hermite>` $\mathbf{MM}^H$ with
+      :py:func:`invertible <sympy.matrices.random.orthogonal>` $\mathbf{M}$
+
+
+In addition to the type of matrix, also the type of entries (as a commutative ring with one)
+to be able to define values (**scalar_set**) can be specified,
+from which the value entries (**scalar**) of the basic matrices are randomly generated.
+
+Because the complexity and amount of entries in the generated compound matrices
+in addition to the **scalar_set**, also by the number of base matrices multiplied for generation
+This can be set using the argument **length**.
+
+Normal forms as well as conjugate types have the arg **spec** to provide a spectrum of eigenvalues.
+
+The product of *elementary* matrices
+
+    >>> from sympy.matrices.random import elementary
+
+    >>> A = elementary(3, (0, 2), -2)
+    >>> A
+    Matrix([
+    [1, 0, -2],
+    [0, 1,  0],
+    [0, 0,  1]])
+
+    >>> B = elementary(3, (1, 2), 1)
+    >>> B
+    Matrix([
+    [1, 0, 0],
+    [0, 1, 1],
+    [0, 0, 1]])
+
+    >>> C = elementary(3, (1, 0), -1)
+    >>> C
+    Matrix([
+    [ 1, 0, 0],
+    [-1, 1, 0],
+    [ 0, 0, 1]])
+
+gives an *invertible* matrix $\mathbf{M}$
+
+    >>> M = A*B*C
+    >>> M
+    Matrix([
+    [ 1, 0, -2],
+    [-1, 1,  1],
+    [ 0, 0,  1]])
+
+    >>> M.inv()
+    Matrix([
+    [1, 0, 2],
+    [1, 1, 1],
+    [0, 0, 1]])
+
+
+Random Generator
+----------------
+
+The module uses its own random generator ``rand`` which is a instance of ``random.Random()``.
+Hence, the state can be set by invoking ``rand.seed()``.
+
+
+    >>> from sympy.matrices.random import rand, orthogonal
+    >>> rand.seed(1)
+    >>> orthogonal(3)
```

Comment:
I think it would be better if there was a way of seeding that didn't require global state. For example maybe a function like `orthogonal` should be able to take a seed or a RNG as a parameter like:
```python
>>> from random import Random
>>> rng = Random()
>>> orthogonal(3, seed=rng)
>>> orthogonal(3, seed=1)
```
We should probably try to make this consistent with other random functions in sympy if possible. There has been a lot of discussion about how to do seeding in the stats module.

When the user does not supply a seed then I think that sympy should have a centralised rng to use for all things. There are long open issues about this - I'm not proposing that you need to go around and fix those issues across the codebase but rather than adding a module-specific seed I think it would be better to add a centralised seed for all of sympy and use that here.